src/components/post/newPost/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { Fragment, useState, useEffect } from 'react';
+
+// Components
+import NewPostDisplay from './NewPostDisplay';
+
+// MUI
+import withStyles from '@material-ui/core/styles/withStyles';
+import Button from '@material-ui/core/Button';
+import style from '../../../style/style';
+
+// Icons
+import AddIcon from '@material-ui/icons/Add';
+
+// Redux
+import { connect } from 'react-redux';
+import { newUserPost } from '../../../redux/actions/dataActions';
+
+const NewPost = ({ classes = {}, UI = {}, newUserPost }) => {
+  const [open, setOpen] = useState(false);
+  const [error, setError] = useState({});
+  const [postState, setPostState] = useState({
+    body: '',
+  });
+
+  const { body } = postState;
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const handleChange = event => {
+    const { name, value } = event.target;
+    setPostState({
+      [name]: value,
+    });
+  };
+  const handleSubmit = event => {
+    event.preventDefault();
+    const userPost = { body };
+    newUserPost(userPost);
+  };
+  useEffect(() => setError(UI.error), [UI.error]);
```

Comment:
Do we need to save error to local state if we're just passing in the external error state?

Reply:
Was using it locally to make sure the error was displaying properly. Got the error handling to work with the redux store properly. Local error handling now removed.